### PR TITLE
feat(search): adding support for `_entityType` filter in the application layer + frontend

### PR DESCRIPTION
**Pull request by gabe-lyons**
Adds support for the _entityType filter, implemented on the Search Service by @iprentic. This filter functions just like any other search filter and removes the need for treating entity filters as a special case. Thanks @iprentic!

Reviewers: @jjoyce0510 @chriscollins3456 @joshuaeilers @iprentic 

image

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/GetQuickFiltersResolver.java [hunk 1]
```diff
@@ -42,7 +44,7 @@ public class GetQuickFiltersResolver implements DataFetcher<CompletableFuture<Ge
 
   private static final String PLATFORM = "platform";
   private static final int PLATFORM_COUNT = 5;
-  private static final String ENTITY = "entity";
+  private static final String LEGACY_ENTITY_FILTER_NAME = "entity";
```
Comment (chriscollins3456) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/GetQuickFiltersResolver.java:47]: are you using this anywhere?

File: datahub-web-react/src/app/search/AdvancedSearchFilterConditionSelect.tsx [hunk 2]
```diff
@@ -91,7 +90,6 @@ export const AdvancedSearchFilterConditionSelect = ({ filter, onUpdate }: Props)
                     }
                 }}
                 size="small"
-                disabled={filter.field === ENTITY_FILTER_NAME}
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/search/AdvancedSearchFilterConditionSelect.tsx:94]: amazing

File: datahub-web-react/src/app/search/SimpleSearchFilters.tsx [hunk 3]
```diff
@@ -42,7 +45,9 @@ export const SimpleSearchFilters = ({ facets, selectedFilters, onFilterSelect, l
         onFilterSelect(newFilters);
     };
 
-    const sortedFacets = cachedProps.facets.sort((facetA, facetB) => {
+    const filteredFacets = cachedProps.facets.filter((facet) => FILTERS_TO_EXCLUDE.indexOf(facet.field) === -1);
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/search/SimpleSearchFilters.tsx:48]: minor nit: could just use `.includes` here and that will be easier to read than `indexOf === -1`

File: datahub-web-react/src/app/search/SimpleSearchFilters.tsx [hunk 4]
```diff
@@ -2,8 +2,11 @@ import * as React from 'react';
 import { useEffect, useState } from 'react';
 import { FacetFilterInput, FacetMetadata } from '../../types.generated';
 import { SimpleSearchFilter } from './SimpleSearchFilter';
+import { ENTITY_FILTER_NAME, ENTITY_INDEX_FILTER_NAME, LEGACY_ENTITY_FILTER_NAME } from './utils/constants';
 
-const TOP_FILTERS = ['degree', 'entity', 'platform', 'tags', 'glossaryTerms', 'domains', 'owners'];
+const TOP_FILTERS = ['degree', ENTITY_FILTER_NAME, 'platform', 'tags', 'glossaryTerms', 'domains', 'owners'];
+
+const FILTERS_TO_EXCLUDE = [LEGACY_ENTITY_FILTER_NAME, ENTITY_INDEX_FILTER_NAME];
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/search/SimpleSearchFilters.tsx:9]: so we're still getting these fields back in `facets`?

File: datahub-web-react/src/app/search/utils/useFilters.ts [hunk 5]
```diff
@@ -6,6 +6,13 @@ import { FacetFilterInput, FilterOperator } from '../../../types.generated';
 import { decodeComma } from '../../entity/shared/utils';
 import { URL_PARAM_SEPARATOR } from './filtersToQueryStringParams';
 
+function ifLegacyFieldNameTranslate(fieldName) {
+    if (fieldName === 'entity') {
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/search/utils/useFilters.ts:10]: nit: can use constant here

**Review by chriscollins3456 — COMMENTED** [review]
This looks good to me! I have some small nits but nothing big on your side.

I did discover that the entity filter does not work with entity types that have an underscore in them (ie. it works for DATASET and CHART, but not DATA_JOB or DATA_PRODUCT).

Before we merge this PR and make that type of filtering the default I think we need to figure that out. I let @iprentic know as well

File: datahub-web-react/src/app/search/SimpleSearchFilters.tsx (hunk 4 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/SimpleSearchFilters.tsx:9]: Yep- I didn't want to include a breaking change as part of this PR. We can delete it later.

**Pull request merged**